src/services/fetchers/UniswapV3MultiFetcher.ts

```diff
@@ -0,0 +1,117 @@
+import {StaticJsonRpcProvider} from '@ethersproject/providers';
+import {inject, injectable} from 'inversify';
+import {Logger} from 'winston';
+
+import {PoolRepository} from '../../repositories/PoolRepository.js';
+import {DexProtocolName} from '../../types/DexProtocolName.js';
+import {ChainsIds} from '../../types/ChainsIds.js';
+import {ContractHelperRepository} from '../../repositories/ContractHelperRepository.js';
+import {BigNumber} from 'ethers';
+
+export type UniswapV3MultiFetcherParams = {
+  chainFrom: string[];
+  token0: string;
+  token1: string;
+};
+
+export interface OutputValues {
+  token0: string;
+  token1: string;
+  value: number;
+}
+
+@injectable()
+class UniswapV3MultiFetcher {
+  provider: StaticJsonRpcProvider | undefined;
+  logPrefix = '[UniswapV3MultiFetcher]';
+  readonly dexProtocol = DexProtocolName.UNISWAP_V3;
+  readonly chainId = ChainsIds.ETH;
+
+  @inject('Logger') protected logger!: Logger;
+  @inject(PoolRepository) protected poolRepository!: PoolRepository;
+  @inject(ContractHelperRepository) contractRepository!: ContractHelperRepository;
+
+  async apply(inputs: UniswapV3MultiFetcherParams[]): Promise<OutputValues[]> {
+    this.logger.debug(`${this.logPrefix}: start with inputs ${JSON.stringify(inputs)}`);
+
+    const poolsToFetch = await this.getPoolsToFetch(inputs);
+
+    if (poolsToFetch.length === 0) {
+      this.logger.error(`${this.logPrefix} no data to fetch`);
+      return [];
+    }
+
+    const prices = await this.fetchData(poolsToFetch);
+
+    return prices;
+  }
+
+  protected async getPoolsToFetch(inputs: UniswapV3MultiFetcherParams[]) {
+    const poolsToFetch: {
+      pools: string[];
+      base: string;
+      quote: string;
+    }[] = [];
+
+    for (const input of inputs) {
+      const {chainFrom, token0, token1} = input;
+      this.logPrefix = '[UniswapV3MultiFetcher]';
```

Comment:
this could be set in the constructor if it doesn't change

Reply:
Done